src/G11n/Currency.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\G11n;
+
+use App\Exceptions\InvalidCurrencyException;
+
+final class Currency
+{
+    private const ALLOWED_VALUES = [
+        'EUR',
+        'USD',
+    ];
+    private $value;
+
+    public function __construct(array $currency)
+    {
+        foreach ($currency as $val) {
+            if (!in_array($val, self::ALLOWED_VALUES)) {
+                throw new InvalidCurrencyException($val);
+            }
+            $this->value = $val;
+        }
+    }
+
+    public function getValue()
+    {
+        return $this->value;
+    }
+
+    public function __toString()
```

Comment:
Return type missing. Also, please do not do magic methods, rather create a proper method with the same name.
```suggestion
    public function toString(): string
```
I don't get why you would provide getValue and toString with the same body 🤔